include/posix/unistd.h

```diff
@@ -11,11 +11,43 @@ extern "C" {
 #endif
 
 #include "sys/types.h"
+#include "sys/stat.h"
+
+#ifdef CONFIG_POSIX_FS
+#include <fs.h>
+
+typedef struct fs_dir_t DIR;
+typedef unsigned int mode_t;
+
+struct dirent {
+	unsigned int d_ino;
+	char *d_name;
```

Comment:
Almost always d_name is a fixed-size array inside struct dirent.  This isn't incorrect per the spec AFAIK, but it's a little weird and it's not impossible that it might break code that does something weird like trying to write over &d_name instead of &d_name[0]